### PR TITLE
rimes handwriting database 

### PR DESCRIPTION
**Pull request by aarora8**
Rimes:
- BPE based sub-word lexicon.
- 6-gram sub-word LM with training text and optional open source French text (64k lines).
- Expanding bounding boxes as augmentation.
- line level decoding but paragraph level scoring.
- same train, test, and validation split as in a recent paper.

Madcat Arabic:
Adding scripts for creation of LM from Arabic gigaword extra corpus text

### CODE REVIEW

**Comment by danpovey** [comment]
@jyang126jhu would you mind checking that this runs for you?  Please comment if you notice any typos or other mistakes.

File: egs/rimes/v1/local/chain/tuning/run_cnn_e2eali_1a.sh [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+  relu-batchnorm-dropout-layer name=tdnn2 input=Append(-4,0,4) dim=$tdnn_dim $tdnn_opts dropout-proportion=0.0
+  relu-batchnorm-dropout-layer name=tdnn3 input=Append(-4,0,4) dim=$tdnn_dim $tdnn_opts dropout-proportion=0.0
+
+  ## adding the layers for chain branch
+  relu-batchnorm-layer name=prefinal-chain dim=$tdnn_dim target-rms=0.5 $tdnn_opts
+  output-layer name=output include-log-softmax=false dim=$num_targets max-change=1.5 $output_opts
+
+  # adding the layers for xent branch
+  # This block prints the configs for a separate output that will be
+  # trained with a cross-entropy objective in the 'chain' mod?els... this
```
Comment (jyang126jhu) [inline egs/rimes/v1/local/chain/tuning/run_cnn_e2eali_1a.sh]: typo: mod?els => models

File: egs/rimes/v1/local/chain/tuning/run_cnn_e2eali_1b.sh [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+  relu-batchnorm-dropout-layer name=tdnn2 input=Append(-4,0,4) dim=$tdnn_dim $tdnn_opts dropout-proportion=0.0
+  relu-batchnorm-dropout-layer name=tdnn3 input=Append(-4,0,4) dim=$tdnn_dim $tdnn_opts dropout-proportion=0.0
+
+  ## adding the layers for chain branch
+  relu-batchnorm-layer name=prefinal-chain dim=$tdnn_dim target-rms=0.5 $tdnn_opts
+  output-layer name=output include-log-softmax=false dim=$num_targets max-change=1.5 $output_opts
+
+  # adding the layers for xent branch
+  # This block prints the configs for a separate output that will be
+  # trained with a cross-entropy objective in the 'chain' mod?els... this
```
Comment (jyang126jhu) [inline egs/rimes/v1/local/chain/tuning/run_cnn_e2eali_1b.sh]: typo: mod?els => models

File: egs/rimes/v1/local/chain/tuning/run_cnn_e2eali_1d.sh [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+  relu-batchnorm-dropout-layer name=tdnn2 input=Append(-4,0,4) dim=$tdnn_dim $tdnn_opts dropout-proportion=0.0
+  relu-batchnorm-dropout-layer name=tdnn3 input=Append(-4,0,4) dim=$tdnn_dim $tdnn_opts dropout-proportion=0.0
+
+  ## adding the layers for chain branch
+  relu-batchnorm-layer name=prefinal-chain dim=$tdnn_dim target-rms=0.5 $tdnn_opts
+  output-layer name=output include-log-softmax=false dim=$num_targets max-change=1.5 $output_opts
+
+  # adding the layers for xent branch
+  # This block prints the configs for a separate output that will be
+  # trained with a cross-entropy objective in the 'chain' mod?els... this
```
Comment (jyang126jhu) [inline egs/rimes/v1/local/chain/tuning/run_cnn_e2eali_1d.sh]: typo: mod?els => models

**Comment by jyang126jhu** [comment]
I have run the "run_end2end.sh", I didn't find any error during this test, and the results are as follow:
# System                      cnn_e2eali_1a
# WER                              7.80
# CER                              2.72
# WER val                          9.18
# CER val                          3.31
# Final train prob              -0.0721
# Final valid prob              -0.0913
# Final train prob (xent)       -0.7290
# Final valid prob (xent)       -0.8351
# Parameters                      4.95M

**Comment by aarora8** [comment]
Thank you, yeah the result is very close to the result I got (%WER 7.91 [ 446 / 5639, 53 ins, 54 del, 339 sub ] exp/chain/cnn_e2eali_1a/decode_test/wer_8_1.0). @jyang126jhu mentioned a modification in the review... I will quickly do it.

**Comment by danpovey** [comment]
Thanks, Ashish and Jinyi! 
I don't have a lot of time to review this in detail but I am merging it.

**Pull request merged**